scrapy/core/downloader/__init__.py

```diff
@@ -59,6 +59,8 @@ def __str__(self):
 def _get_concurrency_delay(concurrency, spider, settings):
     delay = settings.getfloat('DOWNLOAD_DELAY')
     if hasattr(spider, 'download_delay'):
+        warnings.warn("The Spider.download_delay attribute is deprecated",
+                      category=ScrapyDeprecationWarning, stacklevel=2)
```

Comment:
`warnings` and `ScrapyDeprecationWarning` are not imported in this file, according to the flake8 test. Also, please add a test that verifies that this warning is indeed logged; there should be a few existing, similar tests to serve as inspiration.

Reply:
Hi @GeniusLearner

To do it, it should be as easier as doing:

```python
with warnings.catch_warnings(record=True) as w:
    <whatever you need to do>

    self.assertEqual(len(w), 1)
    self.assertIn("Ignoring invalid sitemap", w[0].message)
```

You can also use `self.assertWarns` and `self.assertWarnsRegex`.